### PR TITLE
systemctl: is-failed: check if system is degraded when no unit given

### PR DESCRIPTION
**Pull request by YHNdnzj**
Closes https://github.com/systemd/systemd/issues/3335

### CODE REVIEW

File: src/systemctl/systemctl-is-active.c [hunk 1]
```diff
@@ -38,8 +38,8 @@ static int check_unit_generic(int code, const UnitActiveState good_states[], int
                 if (!arg_quiet)
                         puts(unit_active_state_to_string(active_state));
 
-                for (int i = 0; i < nb_states; ++i)
-                        if (good_states[i] == active_state) {
+                FOREACH_ARRAY(good_state, good_states, nb_states)
```
Comment (poettering) [inline src/systemctl/systemctl-is-active.c:41]: if you touch this, maybe change nb_states type to size_t, as is appropriate for an array size

File: man/systemctl.xml [hunk 2]
```diff
@@ -219,15 +219,14 @@ Sun 2017-02-26 20:57:49 EST  2h 3min left  Sun 2017-02-26 11:56:36 EST  6h ago
...
 
           <listitem>
-            <para>Check whether any of the specified units are in a
-            "failed" state. Returns an exit code
-            <constant>0</constant> if at least one has failed,
-            non-zero otherwise. Unless <option>--quiet</option> is
-            specified, this will also print the current unit state to
-            standard output.</para>
+            <para>Check whether any of the specified units is in the "failed" state. If no unit is specified,
+            check whether there're failed units, which corresponds to the <literal>degraded</literal> state
```
Comment (poettering) [inline man/systemctl.xml]: there're -> there are any

File: src/systemctl/systemctl-is-active.c [hunk 3]
```diff
@@ -69,5 +69,30 @@ int verb_is_failed(int argc, char *argv[], void *userdata) {
...
+        _cleanup_free_ char *state = NULL;
+        sd_bus *bus;
+
+        r = acquire_bus(BUS_MANAGER, &bus);
+        if (r < 0)
+                return r;
+
+        r = bus_get_property_string(bus, bus_systemd_mgr, "SystemState", &error, &state);
+        if (r < 0) {
+                log_error_errno(r, "Failed to query system state: %s", bus_error_message(&error, r));
```
Comment (poettering) [inline src/systemctl/systemctl-is-active.c]: just do `return log_error_errno()` here?

File: src/systemctl/systemctl.c [hunk 4]
```diff
@@ -162,7 +162,7 @@ static int systemctl_help(void) {
                "  list-timers [PATTERN...]            List timer units currently in memory,\n"
                "                                      ordered by next elapse\n"
                "  is-active PATTERN...                Check whether units are active\n"
-               "  is-failed PATTERN...                Check whether units are failed\n"
+               "  is-failed [PATTERN...]              Check whether units are failed\n"
```
Comment (poettering) [inline src/systemctl/systemctl.c]: adjust the text on the right?

**Comment by poettering** [comment]
looks good

**Review by bluca — APPROVED** [review]

**Pull request merged**